pkg/trustedresources/verify.go

```diff
@@ -63,19 +63,21 @@ type VerificationResult struct {
 	Err error
 }
 
-// VerifyTask verifies the signature and public key against task.
+// VerifyResource verifies the signature and public key against resource (v1beta1 task and pipeline).
 // VerificationResult is returned with different types for different cases:
 // 1) Return VerificationResult with VerificationSkip type, when no policies are found and no-match-policy is set to ignore
 // 2) Return VerificationResult with VerificationPass type when verification passed;
 // 3) Return VerificationResult with VerificationWarn type, when no matching policies and feature flag "no-match-policy" is "warn", or only Warn mode verification policies fail. Err field is filled with the warning;
 // 4) Return VerificationResult with VerificationError type when no policies are found and no-match-policy is set to fail, the resource fails to pass matched enforce verification policy, or there are errors during verification. Err is filled with the err.
-// refSource contains the source information of the task.
-func VerifyTask(ctx context.Context, taskObj *v1beta1.Task, k8s kubernetes.Interface, refSource *v1beta1.RefSource, verificationpolicies []*v1alpha1.VerificationPolicy) VerificationResult {
```

Comment:
Wondering if there is a way to reserve refactored functions like what we did for deprecated fields if that would be better? ie. func VerifyTask(..., task, ...) {VerifyResource(..., task, ...)} ?

Reply:
That's a good point, 🤔 Let me try that

Reply:
Added